### PR TITLE
UX: Multichain: Create Circle CI job for MULTICHAIN feature flag

### PR DESCRIPTION
**Pull request by darkwing**
## **Description**
This PR would add a job to run E2E tests for the MULTICHAIN environment variable.  Doing so will ensure we update tests as PRs are merged and not simply updating all of them when we remove the flag.

NOTE: Send flow for Multichain is WIP hence the tests involving send would always be true for now and we will update it once the new send flow is ready.

## **Related issues**

_Fixes https://github.com/MetaMask/MetaMask-planning/issues/1466

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained:
  - [ ] What problem this PR is solving.
  - [ ] How this problem was solved.
  - [ ] How reviewers can test my changes.
- [ ] I’ve indicated what issue this PR is linked to: Fixes #???
- [ ] I’ve included tests if applicable.
- [ ] I’ve documented any added code.
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)).
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by darkwing** [comment]
@kumavis @brad-decker I'd like to add this build so that we can run E2Es on `MULTICHAIN`-specific PRs so that we're forced to updated existing E2Es during our feature flagged design changes.  The alternative would be, once the `MULTICHAIN` design changes are approved and the flag is ready to be removed, we'll need to updated dozens of E2Es.  It wasn't pleasant the first time, I'd prefer not to do it again!

**Review by vthomas13 — DISMISSED** [review]

**Review by brad-decker — DISMISSED** [review]

**Comment by brad-decker** [comment]
Just FYI I believe this setup will result in the test running in CI but not being **required** in GitHub because you didn't make the all-tests-pass job depend on the new suite. IF thats intention remember you'll need to manually hold merging of your PRs until the test passes (or make an executive decision to merge with failing tests and do follow ups) i'm sure you knew this just making sure its explicit.

**Comment by brad-decker** [comment]
Also, what is the plan for firefox?

**Comment by darkwing** [comment]
> Also, what is the plan for firefox?

I'll be adding Firefox when we get closer to release.  With the Firefox flakiness issues CI experiences, I think it could increase the problem intermittents.

**Comment by darkwing** [comment]
This PR is failing for the right reasons -- the UI changed!  Will fix this shortly!

**Comment by NidhiKJha** [comment]
Some of the mv3 tests are gonna be updated via [https://github.com/MetaMask/metamask-extension/pull/21439](https://github.com/MetaMask/metamask-extension/pull/21439)

**Review by vthomas13 — DISMISSED** [review]

**Review by pedronfigueiredo — DISMISSED** [review]

**Review by NidhiKJha — DISMISSED** [review]

File: test/e2e/tests/import-flow.spec.js [hunk 1]
```diff
@@ -122,33 +125,12 @@ describe('Import flow @no-mmi', function () {
...
         // Send ETH from inside MetaMask
         // starts a send transaction
-        await driver.clickElement('[data-testid="eth-overview-send"]');
+        await openActionMenuAndStartSendFlow(driver);
         await driver.fill(
           'input[placeholder="Enter public address (0x) or ENS name"]',
           '0x2f318C334780961FB129D2a6c30D0763d9a5C970',
         );
         await driver.fill('.unit-input__input', '1');
-
```
Comment (PeterYinusa) [inline test/e2e/tests/import-flow.spec.js:131]: Why did we remove all the steps below from the test?

File: test/e2e/metamask-ui.spec.js [hunk 2]
```diff
@@ -282,8 +286,14 @@ describe('MetaMask @no-mmi', function () {
...
   describe('Send token from inside MetaMask', function () {
+    if (process.env.MULTICHAIN) {
+      return;
+    }
     it('starts to send a transaction', async function () {
-      await driver.clickElement('[data-testid="eth-overview-send"]');
+      await openActionMenuAndStartSendFlow(driver);
+      if (process.env.MULTICHAIN) {
+        return;
+      }
```
[the remark this answers is not in the document]
Comment (NidhiKJha) [reply test/e2e/metamask-ui.spec.js:296]: Umm this was a bit weird. After clicking the driver on, when we have to fill the input for send it was breaking the tests in multichain. Also, we are working on send flow once we complete that, these tests will be updated with the new send flow test. So, I think we can ignore this in CI for multichain.

**Review by pedronfigueiredo — APPROVED** [review]

**Review by vthomas13 — APPROVED** [review]

**Review by brad-decker — APPROVED** [review]

**Pull request merged**